### PR TITLE
Add microservice quick start template for Azure Spring Apps

### PR DESCRIPTION
**Pull request by moarychan**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

Fixes https://github.com/Azure/Azure-Spring-Apps/issues/28

## Changelog

* Remove the outdated quick start template `quickstarts/microsoft.appplatform/azure-spring-cloud`.
* Add a new subfolder `quickstarts/microsoft.appplatform/spring`.
* Add a quick start template for PetClinic sample deployment, targeted to the Azure Spring Apps enterprise plan.

Try this template in another branch:

[Deploy To Azure](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fmoarychan%2Fazure-quickstart-templates%2Fmoary%2Fupdate-microservice%2Fquickstarts%2Fmicrosoft.appplatform%2Fmicroservice-apps-enterprise-plan%2Fazuredeploy.json)

### CODE REVIEW

**Comment by moarychan** [comment]
Below are two best practices that can not be applied:
- Template Should Not Contain Blanks - Empty property:  {} Line: 196, Column: 28
   ```text
   Required property &#39;properties&#39; not found in JSON. Path &#39;&#39;, line 1, position 2. (Code: BadArgument, Target: properties)
   ```
- IDs Should Be Derived From ResourceIDs - Property: "buildResultId" must use one of the following expressions for an resourceId property
  ```text
  Invalid arguments: properties.source.buildResultId should not be null or empty. (Code: BadArgument)
  ```

File: quickstarts/microsoft.appplatform/microservice-apps-enterprise-plan/README.md [hunk 1]
```diff
@@ -4,24 +4,24 @@ page_type: sample
...
-urlFragment: azure-spring-cloud
+urlFragment: azure-spring-apps
 languages:
 - json
 ---
 # Deploy a simple Azure Spring Apps microservice application
 
-![Azure Public Test Date](https://azurequickstartsservice.blob.core.windows.net/badges/quickstarts/microsoft.appplatform/azure-spring-cloud/PublicLastTestDate.svg)
-![Azure Public Test Result](https://azurequickstartsservice.blob.core.windows.net/badges/quickstarts/microsoft.appplatform/azure-spring-cloud/PublicDeployment.svg)
+![Azure Public Test Date](https://azurequickstartsservice.blob.core.windows.net/badges/quickstarts/microsoft.appplatform/microservice-apps-enterprise-plan/PublicLastTestDate.svg)
```
Comment (saragluna) [inline quickstarts/microsoft.appplatform/microservice-apps-enterprise-plan/README.md:13]: Would it be better if we call this folder microsoft.appplatform/azure-spring-apps/?

File: quickstarts/microsoft.appplatform/microservice-apps-enterprise-plan/README.md [hunk 2]
```diff
@@ -4,24 +4,24 @@ page_type: sample
...
+[![Visualize](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/visualizebutton.svg?sanitize=true)](http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fquickstarts%2Fmicrosoft.appplatform%2Fmicroservice-apps-enterprise-plan%2Fazuredeploy.json)
 
-This template deploys a simple Azure Spring Apps microservice application to run on Azure. You'll see that there are two deployments in the template: The first deployment provisions resources, but traffic is not yet routed to the apps; the second deployment sets the apps to _active_.
+This template deploys microservice applications to the enterprise plan of Azure Spring Apps. For more information about the microservice sample, see [Azure-Samples/spring-petclinic-microservices](https://github.com/Azure-Samples/spring-petclinic-microservices).
```
Comment (saragluna) [inline quickstarts/microsoft.appplatform/microservice-apps-enterprise-plan/README.md:25]:
```suggestion
This template deploys the PetClinic microservice to the Azure Spring Apps enterprise plan. For more information about the microservice sample, see [Azure-Samples/spring-petclinic-microservices](https://github.com/Azure-Samples/spring-petclinic-microservices).
```

File: quickstarts/microsoft.appplatform/microservice-apps-enterprise-plan/scripts/deploy-jar-to-asa.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+set -Eeuo pipefail
```
Comment (saragluna) [inline quickstarts/microsoft.appplatform/microservice-apps-enterprise-plan/scripts/deploy-jar-to-asa.sh:3]: Do we still need this script for the enterprise plan?

**Review by saragluna — CHANGES_REQUESTED** [review]

File: quickstarts/microsoft.appplatform/microservice-apps-enterprise-plan/scripts/deploy-jar-to-asa.sh (hunk 3 above)
Comment (saragluna) [reply quickstarts/microsoft.appplatform/microservice-apps-enterprise-plan/scripts/deploy-jar-to-asa.sh:3]: My mistake, thought this is another file.

File: quickstarts/microsoft.appplatform/microservice-apps-enterprise-plan/README.md (hunk 1 above)
Comment (moarychan) [reply quickstarts/microsoft.appplatform/microservice-apps-enterprise-plan/README.md:13]: It's a common name, and we should add more quickstart template here.

**Comment by alex-frankel** [comment]
Hi @moarychan - we cannot review any PRs until all of the CI tests are passing.

**Comment by azure-quickstarts** [comment]
@moarychan - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@moarychan - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@moarychan - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@moarychan - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@moarychan - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@moarychan - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@moarychan - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@moarychan - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@moarychan - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by moarychan** [comment]
Hi @alex-frankel , could you help confirm the below best practice test case? I would like to make a clarification to see if this check can be skipped; if so, please help to review this PR.

The `Microsoft.AppPlatform/Spring/apps/deployments`'s property `properties.source.buildResultId` can not be analyzed correctly. The actual value looks like `[concat(resourceId('Microsoft.AppPlatform/Spring/buildServices/builds', variables('asaInstanceName'), 'default', concat(variables('adminServerAppName'), '-default')), '/results/1')]`, the default value is `<default>`, which will point to the latest build, such as `xx/build/2`, `xxx/build/3`. 

The `arm-ttk` always says 
```console
IDs Should Be Derived From ResourceIDs - Property: "buildResultId" must use one of the following expressions for an resourceId property:            
extensionResourceId,resourceId,subscriptionResourceId,tenantResourceId,if,parameters,reference,variables,subscription,guid
```

Currently, the ARM template is not convenient for deploying source code, so we prepared the deployment script to use Azure CLI instead of the ARM template deployment ability.

**Comment by azure-quickstarts** [comment]
@moarychan - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@moarychan - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by alex-frankel** [comment]
The arm-ttk checks are optional. Just need to make sure the quickstart deploys successfully, so this can be merged.

**Pull request merged**